torch/_inductor/decomposition.py

```diff
@@ -615,28 +615,6 @@ def randn_like(
     ).to(memory_format=get_like_layout(self, memory_format))
 
 
-@register_decomposition(aten.full_like)
-def full_like(
-    self: torch.Tensor,
-    fill_value: Union[int, float],
-    *,
-    dtype: Optional[torch.dtype] = None,
-    layout: Optional[torch.layout] = None,
-    device: Optional[torch.device] = None,
-    pin_memory: bool = False,
-    requires_grad: bool = False,
-    memory_format: torch.memory_format = torch.preserve_format,
-) -> torch.Tensor:
-    return torch.full(
-        [*self.size()],
-        fill_value,
-        dtype=dtype or self.dtype,
-        layout=layout or self.layout,
-        device=device or self.device,
-        requires_grad=requires_grad,
-    ).to(memory_format=get_like_layout(self, memory_format))
```

Comment:
Instead of adding a prim - would it make sense to expand the decomposition so that we follow the logic here ?

https://github.com/pytorch/pytorch/blob/66ac724b56e6c37a534f3e066423ef2f41d7477f/aten/src/ATen/native/TensorFactories.cpp#L461-L486

```
if self.is_non_overlapping_and_dense():
     return torch.full(...).to(memory_format()
else:
     stride_permutation = infer_dense_strides(self.sizes(), self.strides())
     return self.permute(stride_permutation)
```

What do you think?